### PR TITLE
Bump to ruby 2.7.x

### PR DESCRIPTION
**Pull request by jmarrec**
ref https://github.com/NREL/openstudio-gems/pull/42

### CODE REVIEW

File: openstudio-extension.gemspec
Comment (jmarrec) [inline openstudio-extension.gemspec:33]:
```suggestion
  spec.add_development_dependency 'octokit', '~> 4.18.0' # for change logs
```

If this is only for changelog, this should be a dev dependency

File: openstudio-extension.gemspec
Comment (jmarrec) [inline openstudio-extension.gemspec:31]: Is this still relevant? Should it be a development_dependency instead?

https://github.com/NREL/bcl-gem/issues/8

File: openstudio-extension.gemspec
Comment (nllong) [reply openstudio-extension.gemspec:33]: Just to make sure, dependent gems will install this development dependency? I just want to make sure that it is just not a dev dependency for this gem.

**Comment by nllong** [comment]
Are we expecting these unit tests to pass?

**Review by nllong — APPROVED** [review]
Changes look good, need unit tests to pass to be sure though.

Also need OS workflow and measure tester to be released before this will pass.

File: openstudio-extension.gemspec
Comment (jmarrec) [reply openstudio-extension.gemspec:33]: https://guides.rubygems.org/specification-reference/#add_development_dependency

> Development dependencies aren't installed by default and aren't activated when a gem is required.

**Comment by tijcolem** [comment]
@nllong  Yes, we are expecting unit tests to pass once we have an updated docker container containing ruby 2.7.2 and an updated version of OpenStudio to test against.  Hit an error related to bundler version issues. Should have something up to test soon and then get all tests passed here before the merge.

**Comment by nllong** [comment]
> @nllong Yes, we are expecting unit tests to pass once we have an updated docker container containing ruby 2.7.2 and an updated version of OpenStudio to test against. Hit an error related to bundler version issues. Should have something up to test soon and then get all tests passed here before the merge.

Good deal. I'll review again and merge once the tests are passing!

**Comment by tijcolem** [comment]
Currently failing due to required specs. 

These will need get merge and released before this can get tested and merged.  

https://github.com/NREL/OpenStudio-workflow-gem/pull/127 (passing CI)
https://github.com/NREL/bcl-gem/pull/19 (No CI configured)
https://github.com/NREL/OpenStudio-measure-tester-gem/pull/49 (passing CI)

**Comment by nllong** [comment]
@tijcolem -- I think this is good to go! Let me know if there are any issues and assuming none, I'll release.

**Comment by tijcolem** [comment]
@nllong  Awesome! CI looks happy now and tests are passing.  I'd say release.

**Pull request merged**
